### ISSUE TITLE
Deserialization not working at all

### ISSUE THREAD

**Issue by trevore-q4** [GillianPerard/typescript-json-serializer#141]
I'm very mystified at this issue.

Following your examples for deserialization and no fields are being assigned in the resulting object, everything is undefined.

I created a simple project with very few files to test this behavior:

tsconfig.json
```
{
    "compilerOptions": {
        "noImplicitAny": false,
        "strictPropertyInitialization": false,
        "emitDecoratorMetadata": true,
        "experimentalDecorators": true,
        "baseUrl": ".",
      }
}
```
package.json
```
{
  "name": "ts-node-testing",
  "version": "1.0.0",
  "description": "",
  "main": "app.ts",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "dependencies": {
    "ts-node": "^10.0.0",
    "typescript": "^4.3.2",
    "typescript-json-serializer": "^3.3.0"
  }
}
```

app.ts
```
import { deserialize, JsonProperty, Serializable } from "typescript-json-serializer";

@Serializable()
export class Organization {
    constructor() {}
    @JsonProperty() id: string;
    @JsonProperty() name: string;
}

const json = `{"id":"1","name":"My Name"}`;

const obj = JSON.parse(json);
const org = deserialize<Organization>(json, Organization);

console.log(org.id);
```

You can see in my variable watch window that using JSON.parse results in fields being created, but an empty object is returned from deserialize(): https://i.imgur.com/Pl9Rn4j.png

**Comment by GillianPerard** [comment]
Hi, thank you for trying to use my lib :).

In your example, you use your `json object and not the parsed one `obj`.

Maybe I could add to my lib the autoparsing if the first parameter is a string.

**Comment by trevore-q4** [comment]
Hi Gillian,

Thanks for responding. Yes I was doing that on purpose because I misunderstood and thought I should pass in a raw json string since the parameter is named `json`. I think that would be a useful feature, though.

Cheers.

**Comment by GillianPerard** [comment]
Hi, I published the new 3.4.0 version to support the json string 😊.

**Comment by trevore-q4** [comment]
Cheers!
